lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -286,6 +294,130 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
     end
   end
 
+  def safe_partman_create_parent(table, **options)
+    if options[:retention].present? || options[:retention_keep_table] == false
+      raise PgHaMigrations::UnsafeMigrationError.new(":retention and/or :retention_keep_table => false can potentially result in data loss if misconfigured. Please use unsafe_partman_create_parent if you want to set these options")
+    end
+
+    unsafe_partman_create_parent(table, **options)
+  end
+
+  def unsafe_partman_create_parent(
+    table,
+    key:,
+    interval:,
+    infinite_time_partitions: true,
+    inherit_privileges: true,
+    premake: nil,
+    start_partition: nil,
+    template_table: nil,
+    retention: nil,
+    retention_keep_table: nil
+  )
+    raise ArgumentError, "Expected <key> to be present" unless key.present?
+    raise ArgumentError, "Expected <interval> to be present" unless interval.present?
+
+    if ActiveRecord::Base.connection.postgresql_version < 11_00_00
+      raise PgHaMigrations::InvalidMigrationError, "Native partitioning with partman not supported on Postgres databases before version 11"
+    end
+
+    fully_qualified_table_name = _fully_qualified_partman_name(table)
+
+    create_parent_options = {
+      parent_table: fully_qualified_table_name,
+      control: key,
+      type: "native",
+      interval: interval,
+      premake: premake,
+      start_partition: start_partition,
+    }.compact
+
+    if template_table.present?
+      create_parent_options[:template_table] = _fully_qualified_partman_name(template_table)
+    end
+
+    create_parent_sql = create_parent_options.map { |k, v| "p_#{k} := #{connection.quote(v)}" }.join(", ")
+
+    unsafe_execute("SELECT #{_quoted_partman_schema}.create_parent(#{create_parent_sql})")
```

Comment:
We should probably wrap this in a `say_with_time` block like we used to have to do when we implemented our own check constraint methods: https://github.com/braintree/pg_ha_migrations/pull/84/files#diff-e246218fba61b707b474c288f842dc0bf0c83436253bf2b49d05216d222f034cL181

(and the other commands too)

Reply:
Done